models/magiceden/solana/magiceden_solana_trades.sql

```diff
@@ -0,0 +1,243 @@
+{{
+    config(
+        schema = 'magiceden_solana'
+        , tags = ['dunesql']
+        , alias = alias('trades')
+        ,materialized = 'incremental'
+        ,file_format = 'delta'
+        ,incremental_strategy = 'merge'
+        ,unique_key = ['project','trade_category','outer_instruction_index','inner_instruction_index','account_metadata','tx_id']
+        ,post_hook='{{ expose_spells(\'["solana"]\',
+                                    "project",
+                                    "magiceden",
+                                    \'["ilemi"]\') }}'
+    )
+}}
+
+with 
+    royalty_logs as (
+        with nested_logs as (
+            SELECT 
+                distinct 
+                call_tx_id
+                , call_block_slot
+                , call_outer_instruction_index
+                , call_inner_instruction_index
+                , cast(json_extract_scalar(json_parse(split(logs, ' ')[3]), '$.royalty') as double) as royalty
+                , logs
+            FROM (
+                SELECT
+                    call_tx_id
+                    , call_block_slot
+                    , call_outer_instruction_index
+                    , call_inner_instruction_index
+                    , call_log_messages
+                FROM {{ source('magic_eden_solana','m2_call_mip1ExecuteSaleV2') }}
```

Comment:
this table is relatively large (>750k rows). since this is an incremental model, can we add incremental logic here?

Reply:
alan introduced new incremental macro as well, to keep spellbook clean and ability to alter number of days we look back in one universal location too, rather than each individual model (work to do on our side to get all spells updated, but might as well start clean here).

reference:
https://github.com/duneanalytics/spellbook/pull/4487/files#diff-dc8d4490c16a624e93e14dcb8cae5c6f8f53697c962e2e4fae095432d705eb4cR51-R53